### ISSUE TITLE
Refactor NodeIndex.prepare()

### ISSUE THREAD

**Issue by scottx611x** [refinery-platform/refinery-platform#2717]
* Specific code commit: 99658a792f3a81971e2b6c2be8f406360df32660

[`dataset_manager.search_indexes.NodeIndex.prepare`](https://github.com/refinery-platform/refinery-platform/blob/master/refinery/data_set_manager/search_indexes.py#L97) is getting a bit unruly, but has [nearly 100% test coverage](https://codecov.io/gh/refinery-platform/refinery-platform/src/develop/refinery/data_set_manager/search_indexes.py#L97). 

It's probably worth refactoring a bit.

**Comment by hackdna** [comment]
Yes, file status logic should be simplified and moved to `FileStoreItem.get_import_status()`. Also, file import task(s) should return correct status depending on the actual state of the file import process (waiting to run, in progress, success, failure, etc). See #2359.
